Sources/CacheAdvance/CacheReader.swift

```diff
@@ -42,7 +45,7 @@ final class CacheReader {
     }
 
     /// Returns the next encodable message, seeking to the beginning of the next message.
-    func nextEncodedMessage(previousReadWasEmpty: Bool = false) throws -> Data? {
```

Comment:
The case we previously caught with `previousReadWasEmpty` is now caught by our new checks below!